more_itertools/more.py

```diff
@@ -1411,6 +1413,41 @@ def stagger(iterable, offsets=(-1, 0, 1), longest=False, fillvalue=None):
     )
 
 
+class UnequalIterablesException(Exception):
+    def __init__(self, idxs, names=None):
+        if names is None:
+            depleted = "Indices: {}".format(
+                ", ".join(str(idx) for idx in idxs)
+            )
+        else:
+            depleted = "Columns: {}".format(
+                ", ".join(names[idx] for idx in idxs)
+            )
+        super().__init__(
+            "Iterables have different lengths."
+            "{} depleted before rest.".format(depleted)
+        )
+
+
+def zip_equal(*iterables, names=None):
+    """``zip`` the input *iterables* together, but throw an
+    ``UnequalIterablesException`` if any of the *iterables* terminate before
+    the others. The exception will give the indices which terminated too early.
+    Alternatively *names* can be passed and it will give the name corresponding
+    to the early terminating indices.
+
+    """
+    sentinel = object()
+    if names is not None and len(iterables) != len(names):
+        raise ValueError("Number of iterable and names didn't match")
+
+    for combo in zip_longest(*iterables, fillvalue=sentinel):
+        finished_idxs = [idx for idx, c in enumerate(combo) if c is sentinel]
```

Comment:
Like `names`, I think this is probably too much - let's just go with `if _marker in combo`?

Reply:
I have used "is" instead since it can not be overloaded as __eq__ can, so we know for sure that we actually have _marker rather than a tricky/misbehaving object (_marker is only used with is elsewhere also) but I've removed the rest